### ISSUE TITLE
User interface performance bad when Thingiview running

### ISSUE THREAD

**Issue by MichaelAtOz** [openscad/openscad#1019]
The Thingiverse thingiview application, running in Firefox (v28 atm), brings OpenSCAD UI to its knees, and the whole system too.

I have traced it to GPU utilisation hitting 100%. Something is not handling that well.

This is a problem with Thingiview, but OpenSCAD makes it much worse for some reason.

When thingiview is running (without OpenSCAD) it consumes ~80% of the GPU (& 100% of one CPU core, = 50% on my two core - with lame graphics), not nicely behaved...

When OpenSCAD running (without Thingiview), with a 'new document' - empty, just rotating the view axes, it consumes ~10-15% of the GPU. With a complex model loaded, and noticeable lag then moving the display, it is hitting 80-100% GPU.

So with OpenSCAD with a 'new document', then click on the Thingiview button, it loads the STL then it works OK. Click on OpenSCAD, rotate the display axes and/or click on menus. Initially it is OK, but progressivly slows down, same for all applications. The mouse freezes for some second then catches up, menu selections take seconds to react to clicks. Basically any UI graphics event gets stalled for some seconds, then catches up.

I have tried other graphics intense apps like Netfab, they don't have an effect.

So I tried to find out what was happening.

This shows the GPU (and other info at the top row) with both running. Open it in another tab to view larger.
![gpu](https://cloud.githubusercontent.com/assets/1972961/5060740/175f490c-6dbd-11e4-8a7f-8f5177557b09.JPG)

Basically something seems to cause firefox (Thingiview) to periodically grab 100% GPU, perhaps 'catching up' from OpenSCAD taking a chunk? The system freezes for a few seconds then it processes the backlog, then freezes etc. Remember this is with no model loaded in OpenSCAD.

I suspect that there may be something locking the GPU???

It is also related to the size of the OpenSCAD window, below a certain size (1/4 screen) it doesn't happen. Obviously related to the load on the GPU.

Note that I have heaps of windows/tabs/apps open, so Windows GUI is pretty loaded, I haven't checked yet whether running a lesser load affects this.
##

**Comment by achiestdragon** [comment]
same issue  with nightly build ( 2017.12.03.nightly (git c326079)) see png screenshot 
[https://drive.google.com/file/d/1CHB60sCb61VzavYWvK8neuyUideE9ldc/view?usp=sharing](url)

cpu usage meter in ksysguard , first peak set of 100% usage was a f5 render  the second peak set of cpu usage was a result of moving the mouse over the preview window    without pressing a mouse button  , and has resulted in a couple of app crashes (only noted on this git build)
